### DISCUSSION TITLE
Why in Auto Codec always I got AV1 ? RustDesk 1.2.3 Windows

### DISCUSSION THREAD

**Discussion by mikekgr** [rustdesk/rustdesk#5961] [General]
I realize that every time I connected my Windows 10 PC to a remote Windows 10 PC both running RustDesk 1.2.3 when the Codec is in Auto ALWAYS I got Codec AV1 even if the remote system can use even H265 ? (I have selected H265 and it is working just fine)

Please tell me if I did something wrong.

Thanks

**Comment by 21pages** [comment]
It's not always AV1, but currently AV1 is the top priority option. The following reasons lead to not including H.264/H.265 in the auto codec:
* https://github.com/rustdesk/rustdesk/issues/3434 hwcodec may be slow it's difficult to assess its performance.
* have not add fallback during encoding/decoding

You can choose h265 as default codec in display settings as alternative.

**Reply by MrChenhtlss** [reply]
There seems to be no H265 option in MAC OS version 1.2.3

**Comment by mikekgr** [comment]
> It's not always AV1, but currently AV1 is the top priority option. The following reasons lead to not including H.264/H.265 in the auto codec:
> 
> * [Connected to a specific computer, the remote desktop response is very slow. (hwcodecs slow) #3434](https://github.com/rustdesk/rustdesk/issues/3434) hwcodec may be slow it's difficult to assess its performance.
> * have not add fallback during encoding/decoding
> 
> You can choose h265 as default codec in display settings as alternative.

Dear @21pages  , in my case in all of my PC's running Windows 10 and RustDesk 1.2.3 (10 out of 10) ALL have AV1 as Codec (in Auto selection). In the older days was VP9 now all the time AV1. Is AV1 more advanced or have benefits in comparison to VP9? I don't know why this changed.
Ofcourse I know that I can select any one from the supported in the settings but I am wondering why is stick to AV1

Thanks and Best Regards,
Mike Kranidis

**Comment by deep-soft** [comment]
Until https://github.com/rustdesk/rustdesk/pull/6084
AV1 codec didn't have record capabilities and that's the reason VP9 is default codec for Auto (because VP9 allowed Record).
After that PR (https://github.com/rustdesk/rustdesk/pull/6084) AV1 codec has top priority.
Picture quality is better in AV1 codec mode vs VP9 code mode, in all tests I made.

**Comment by MrChenhtlss** [comment]
The latest 1.2.3 option has H265 but cannot be called

**Reply by deep-soft** [reply]
You must check logs to see if H265 is supported.
See subfolder "check-hwcodec-config" of "log" folder.
